main.py

```diff
@@ -12,9 +15,34 @@
 
 graphs = load_graph_list(filename)
 
-os.makedirs(os.path.dirname('output_files/main/'), exist_ok=True)
+# Test output csv creation
+csv_filepath = create_csv_file('main')
+# Write first row with column names
+write_csv_line(csv_filepath, ['file', 'i', 'time (s)', 'result'])
 
+# Test result accumulators
+end_time_sum = 0
+test_result_passed = True
+
+# main test loop
 for i in range(0, len(graphs)):
     outputfilename = 'output_files/main/' + file + '_' + str(i) + '.dot'
+    os.makedirs(os.path.dirname(outputfilename), exist_ok=True)
+
+    start_time: float = time.time()
     with open(outputfilename, 'w') as g0:
         write_dot(color_refinement(degree_color_initialization(graphs[i])), g0)
+
+    test_result = True
```

Comment:
Ik zie in deze test nergens waar test_result False kan zijn. Is het dan nodig om dit toe te voegen?

Reply:
ik heb een uitgebreidere test toegevoegd, waar ik ook de csv weer lees om de schrijfactie te verifieren.